dot/parachain/provisioner/messages/messages.go

```diff
@@ -0,0 +1,45 @@
+package provisionermessages
+
+import (
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+var (
+	_ Data = (*ProvisionableDataBackedCandidate)(nil)
+	_ Data = (*ProvisionableDataMisbehaviorReport)(nil)
+)
+
+type RequestInherentData struct {
+	RelayParent             common.Hash
+	ProvisionerInherentData chan ProvisionerInherentData
+}
+
+type ProvisionerInherentData struct {
+}
+
+// ProvisionerMessageProvisionableData is a provisioner message.
+// This data should become part of a relay chain block.
+type ProvisionableData struct {
+	RelayParent common.Hash
+	Data        Data
+}
+
+// Data becomes intrinsics or extrinsics which should be included in a future relay chain block.
+type Data interface {
+	IsProvisionableData()
```

Comment:
The name sounds like this should return `bool`.

Reply:
This is a common pattern to express enums in golang, e.g:

```go
type EnumA interface {}
```

Reply:
💡🙏